### DISCUSSION TITLE
Are there plans to add `Config.fix` explicitly to the Python API?

### DISCUSSION THREAD

**Discussion by chpatton013** [ast-grep/ast-grep#1480] [Q&A]
Or maybe a better question is "how can I access the 'fixed' result of a `SgNode.find_all(Config)` call?

The [Patching docs](https://ast-grep.github.io/reference/yaml.html#patching) indicate that there are two (non-experimental) methods to modify matched content: `transform` and `fix`.

[ast_grep_py.Config has a `transform` field](https://github.com/ast-grep/ast-grep/blob/683e398430820549ff5ac947766e829b0c1f7bc4/crates/pyo3/ast_grep_py/__init__.py#L56) (whose content is [documented here](https://ast-grep.github.io/reference/yaml/transformation.html)), but no `fix` field ([documented here](https://ast-grep.github.io/reference/yaml/fix.html)). The [Python bindings in Rust make use of `depythonize()`](https://github.com/ast-grep/ast-grep/blob/683e398430820549ff5ac947766e829b0c1f7bc4/crates/pyo3/src/py_node.rs#L346-L348) to transform the `Config` dictionary into a `SerializableRuleCore` without needing to explicitly handle any specific fields. [That type seems to support the `fix` field](https://github.com/ast-grep/ast-grep/blob/683e398430820549ff5ac947766e829b0c1f7bc4/crates/config/src/rule_core.rs#L59). It seems like it would be easy to pass a value for `Config.fix` (even without updating the Python type shims because everything boils down to dictionaries under the hood).

Problem is: I don't see how to find the result of appling the `fix` to the input `SgNode`.

It's clear that after using either the [`find(Config)`](https://github.com/ast-grep/ast-grep/blob/683e398430820549ff5ac947766e829b0c1f7bc4/crates/pyo3/ast_grep_py/ast_grep_py.pyi#L45-L46) or [`find_all(Config)`](https://github.com/ast-grep/ast-grep/blob/683e398430820549ff5ac947766e829b0c1f7bc4/crates/pyo3/ast_grep_py/ast_grep_py.pyi#L49-L50) overloads that the results of `transform` entries are available [via `get_transformed()`](https://github.com/ast-grep/ast-grep/blob/683e398430820549ff5ac947766e829b0c1f7bc4/crates/pyo3/ast_grep_py/ast_grep_py.pyi#L41), but that function takes a meta-variable name (mapping back to the keys of the `transform` dict) as its input. `Config.fix` doesn't seem to have a relationship to meta-variables, and there's no interface on the `SgNode` or `SgRoot` classes to provide the results of applying the `fix`.

Given an `SgRoot`, I want to apply a bunch of `FixConfig`s and get the transformed result.

**Comment by HerringtonDarkholme** [comment]
There is no plan to do that in PY/JS API since you don't really need transform/fix in a general purpose language.
regular expression and substring like features are well ready in these two languages. 

You may need to generate a patch from a SgNode and apply it.
https://ast-grep.github.io/guide/api-usage/py-api.html#fix-code
